Classes/EventListener/NewsLayoutListener.php

```diff
@@ -0,0 +1,208 @@
+use TYPO3\CMS\Core\Imaging\IconFactory;
+use TYPO3\CMS\Core\Localization\LanguageService;
+use TYPO3\CMS\Core\Page\PageRenderer;
+use TYPO3\CMS\Core\Type\Bitmask\Permission;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+final class NewsLayoutListener
+{
+    public function __construct(
+        protected readonly PageRenderer $pageRenderer,
+        protected readonly IconFactory $iconFactory,
+        protected readonly ExtensionConfiguration $extensionConfiguration
+    ) {}
+
+    public function __invoke(ModifyPageLayoutContentEvent $event): void
+    {
+        if(!$this->extensionConfiguration->get('newspage', 'layout_edit_mode')) {
+            return;
+        }
+
+        $queryParams = $event->getRequest()->getQueryParams();
+        $pageId = (int)($queryParams['id'] ?? 0);
+        $language = (int)($queryParams['language'] ?? 0);
+        $function = (int)($queryParams['function'] ?? 1);
+        $pageInfo = BackendUtility::readPageAccess($pageId, $this->getBackendUser()->getPagePermsClause(Permission::PAGE_SHOW));
+
+        // Display page property inline edit only for doktype=24 and function=1 (layout mode)
+        if($function !== 1 || $pageInfo['doktype'] !== 24 || !$this->isPageEditable($language, $pageInfo)) {
+            return;
+        }
+
+        /** @var NodeFactory $nodeFactory */
+        $nodeFactory = GeneralUtility::makeInstance(NodeFactory::class);
+        $formResultCompiler = GeneralUtility::makeInstance(FormResultCompiler::class);
+        $formDataGroup = GeneralUtility::makeInstance(TcaDatabaseRecord::class);
+
+        if ($language > 0) {
+            $overlayRecord = $this->getLocalizedPageRecord($language, $pageId);
+            $pageId = $overlayRecord['uid'];
+        }
+
+        $formDataCompiler = GeneralUtility::makeInstance(FormDataCompiler::class, $formDataGroup);
+        $formDataCompilerInput = [
+            'tableName' => 'pages',
+            'command' => 'edit',
+            'vanillaUid' => $pageId,
+        ];
+
+        // Render only the palette "tx_newspage_layout" in page layout view:
+        $GLOBALS['TCA']['pages']['types']['24']['showitem'] = '--palette--;;tx_newspage_layout';
+
+        $formData = $formDataCompiler->compile($formDataCompilerInput);
+        $formData['renderType'] = 'fullRecordContainer';
+
+        $formResult = $nodeFactory->create($formData)->render();
+        $formResultCompiler->mergeResult($formResult);
+        $formResultCompiler->printNeededJSFunctions();
+
+        $params = [
+            'returnUrl' => $event->getRequest()->getAttribute('normalizedParams')->getRequestUri(),
+            'edit' => [
+                'pages' => [
+                    $pageId => 'edit',
+                ],
+            ],
+        ];
+
+        /** @var UriBuilder $uriBuilder */
+        $uriBuilder = GeneralUtility::makeInstance(UriBuilder::class);
+        $editUri = (string)$uriBuilder->buildUriFromRoute('newspage_layout_edit', $params);
```

Comment:
What is the reason for using an own endpoint here? Could we just use `tce_db` which forwards to the `SimpleDataHandlerController`?

Reply:
Removed the endpoint and using the SimpleDataHandlerController directly.